cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -429,6 +451,15 @@ private void storeActiveTunnel(final TunnelConfig config, boolean isUdpSupported
       proxyConfig.put("port", config.proxy.port);
       proxyConfig.put("password", config.proxy.password);
       proxyConfig.put("method", config.proxy.method);
+
+      if (config.proxy.prefix != null) {
+        char[] chars = new char[config.proxy.prefix.length];
+        for (int i = 0; i < config.proxy.prefix.length; i++) {
+          chars[i] = (char)(config.proxy.prefix[i] & 0xFF);
```

Comment:
Isn't proxy.prefix a byte array? Why do you need the `& 0xFF`?

Reply:
Otherwise, the Java cast will interpret 0x80 as -128 and sign-extend it into 0xFF80.  I added a comment.